### ISSUE TITLE
Create a Welcome easy-FDM for beginners.

### ISSUE THREAD

**Issue by dany93** [c172p-team/c172p-detailed#1114]
Follows https://github.com/c172p-team/c172p-detailed/issues/1112#issuecomment-392001045
and https://github.com/c172p-team/c172p-detailed/issues/1113#issue-325643076
particularly
https://github.com/c172p-team/c172p-detailed/issues/1112#issuecomment-391721130
https://github.com/c172p-team/c172p-detailed/issues/1112#issuecomment-391528826
https://github.com/c172p-team/c172p-detailed/issues/1112#issuecomment-391797365

**Comment by edward-dauvergne** [comment]
I strongly suggest that you discuss this on the FlightGear mailing list!  I think you'll find not much support for this idea within the community.  I remember such suggestions being dismissed before.  I can't find all such discussions (due to the loss of the Gmane archive), but one I remember is:

- https://www.mail-archive.com/flightgear-devel@lists.sourceforge.net/msg19727.html

I personally think that this is a bad idea.  If someone wants a simple aircraft, they can fly one of the fictional craft out there (OGEL, Bluebird with UFO FDM, etc.).  This concept should not be confused with what looks like a [simple and recent FDM bug](https://sourceforge.net/p/flightgear/mailman/message/36325970/) to me.

Regards,

Edward

**Comment by dany93** [comment]
This does not close the subject.
It will contain a tuning factor which (I hope, please, @wlbragg ?) will allow experienced pilots to tune it by themselves and propose a value. Hopefully, if enough pilots participate, we can have a correct value.
From here or from the mailing list (which shouldn't be the best place)

I'm not sure I will find better support from the mailing list (in addition to polluting this list with a c172 subject). 
Here or from the forum I get responses from people who seem to have a partial knowledge of the problem. They seem to ignore the importance (existence?) of the spiraling propwash effect, were not surprised that this effect has not been taken into account for 20 years of FG.

Whatever the procedure, I hope that the introduction of this tuning coefficient will help for getting test and feedback from real c172 pilots.

Here, fortunately, we have @tigert often giving opinions after tests, and advices. But we miss such feedbacks and accurate data.

From my experience on RL ultralights, the spiraling prowash is sufficient to drive you out off the runway well befrore takeoff if you don't counteract with left rudder.

**Comment by edward-dauvergne** [comment]
Is this FDM change listed as a feature for any of the releases here?  I've always used a little rudder, but all of a sudden I now need a lot.  I suggest discussing on the mailing list where you will find a lot of real pilots who have trained in Cessnas.

**Comment by tonghuix** [comment]
@dany93 I tested 2016.1 and 2017.1 for FDM, feedback could see #1112

**Comment by dany93** [comment]
@tonghuix
Given the importance that it takes, I'm asking for real-pilots feedback. Are you?

In addition, in [this message](https://github.com/c172p-team/c172p-detailed/issues/1112#issuecomment-391291575) you seem to ignore the importance (existence?) of the spiraling propwash effect. Whereas I've read several times that it is more important than the p-factor (particularly for tricycles).
In [this message](https://github.com/c172p-team/c172p-detailed/issues/1112#issuecomment-391389088) I ask you for some more explanations.
Unless that's me and I have not understood something. But I'm still waiting for your response.

**Comment by dany93** [comment]
@edward-dauvergne wrote

> I suggest discussing on the mailing list where you will find a lot of real pilots who have trained in Cessnas.

I'm surprised by this suggestion. I thought that there is enough on the mailing list with the core development.

This feature has been committed on[ 9 Feb 2018](https://github.com/c172p-team/c172p-detailed/commit/eccd4b017237dcacbe5155552d833a552e98f9fe).
I understand that everyone has his own occupations, but, the c172p being the FG flagship, I regret that so many real pilots never responds to our requests for feedback.

**Comment by dany93** [comment]
@wlbragg 
I'm not sure how it works with the options, but I hope, if possible, that the `aero/coefficient/spiral-propwash-coeff` can be manually set in the Properties (in sim). It would enable real pilots who accept testing and give some time to easily try any value (more likely between 0 and 1). Hopefully, this feedback should help converging to a consensus based on reliable opinions.

**Comment by wkitty42** [comment]
On 05/25/2018 11:04 AM, dany93 wrote:
> I understand that everyone has his own occupations, but, the c172p being the
> FG flagship, I regret that so many real pilots never responds to our requests
> for feedback.

FWIW: this repo, the FG forums and the FG dev mailing list have poor visibility 
to real c172p pilots... only those that know about FG would see the requests IF 
they even visit any of those three places... we all know there's a lot of FG 
users that have never been by the forums, much less this repo or the dev mailing 
list...

**Comment by wlbragg** [comment]
> I hope, if possible, that the aero/coefficient/spiral-propwash-coeff can be manually set in the Properties (in sim). 

@dany93 yes, it would be controls/flight/aero/coefficient/spiral-propwash-coeff and it should be accessible.

**Comment by dany93** [comment]
>  it would be controls/flight/aero/coefficient/spiral-propwash-coeff and it should be accessible.

Good. Thank you
But I'd say `fdm/jsbsim/aero/coefficient/spiral-propwash-coeff`
This is not a control.

**Comment by edward-dauvergne** [comment]
> I'm surprised by this suggestion. I thought that there is enough on the mailing list with the core development.

I'm bizarrely surprised that you are surprised!  Lots of aircraft development questions belong on that list.  Especially important FDM developments.  And we are talking about a new "game-like" feature that some core developers may strongly object to, as they have in the past.

The [original C172P](http://wiki.flightgear.org/index.php?title=Cessna_172P&oldid=64944) was created largely by [David Megginson](http://www.megginson.com/).  He is a [real pilot](http://www.megginson.com/bio/).  From memory, he has flown the real thing.  David is a massive contributor of FlightGear core code and aircraft from very early on.  And he is still partly active on the mailing list.

Also Torsten owns a Seneca II.  Stuart [owns and flies a microlight](http://home.flightgear.org/contributors/stuart-buchanan/).  I know many other code developers are real pilots, and there are lots of other real pilots subscribed to the list.

**Comment by onox** [comment]
> Is this FDM change listed as a feature for any of the releases here? I've always used a little rudder, but all of a sudden I now need a lot.

Yes. See the "spiraling propwash effect" in https://github.com/c172p-team/c172p-detailed/releases/tag/version%2F2018.1. Merged to FGDATA by @stuartbuchanan.

**Comment by tonghuix** [comment]
@dany93 sorry that in #1112 , i missed your question that I am *not a real* pilot. But I had a experience that fly a real C172R in a event 12 years ago. And I know some friends who are real pilot, asked them helping test, see #1060

**Comment by onox** [comment]
Also see the [comment](https://github.com/c172p-team/c172p-detailed/issues/947#issuecomment-360999442) by @tigert.

> Or the alternative
that James suggested and I also elaborated on of having a copy of the
git repository mirrored at SourceForge and the aircraft removed from
FGData.  I.e. equivalent to fork #36, as there are currently 35 forks
of the repository on GitHub, but used as a read-only mirror hosted
alongside the FlightGear core infrastructure and used in Torsten's
automated bundling procedure.  This is a very, very easy solution that
removes a lot of the current disjoint between the c172p-detailed team
and the FG core.

@edward-dauvergne If you can fully automate this, then I've no problems with it. Just FYI I think SF is a product from the 90s that does/did [shady](https://en.wikipedia.org/wiki/SourceForge#Controversies) things. So I'm not gonna/can't commit to FGDATA.

**Comment by tonghuix** [comment]
Yes I agree @onox , the SF is too old to adopting for 'modern' open source product development.

**Comment by wlbragg** [comment]
@dany93  sorry, I didn't actually go into the code, I was looking at the diff and picked up the wrong tags, yes it is...
fdm/jsbsim/aero/coefficient/spiral-propwash-coeff
~~~
<fdm>
...
        <jsbsim>
           ....
            <aero>
                <coefficient>
                    <spiral-propwash-coeff type="double">1</spiral-propwash-coeff>
                </coefficient>
            </aero>
        </jsbsim>
</fdm>
~~~
Answer remains the same, yes that should be accessible.

**Comment by wkitty42** [comment]
On 05/25/2018 02:47 PM, onox wrote:
> Just FYI I think SF is a product from the 90s that does/did shady 
> <https://en.wikipedia.org/wiki/SourceForge#Controversies> things. So I'm not 
> gonna/can't commit to FGDATA.

SF is under totally new management and is no longer doing those nasty things 
that the previous management group(s) did...

**Comment by onox** [comment]
I know, but here in NL we say trust comes by foot and leaves on a horse :stuck_out_tongue:

**Comment by wkitty42** [comment]
On 05/25/2018 03:01 PM, onox wrote:
> I know, but here in NL we say trust comes by foot and leaves on a horse 😛


hahaha, yeah... several NL'ers have told me the same over the last 30+ years... 
the thing is, it has been several years since SF was returned to ""proper"" 
activities... one cannot (re)gain their trust if one doesn't use SF's 
services... they offer a whole lot more than repositories (in several formats) 
and issue tracking... github and gitlab don't even come close ;)

anyway, EOT from my side...

**Comment by onox** [comment]
Well, personally I don't like any of the two solutions. I prefer 100 % realistic FDM (that includes complex procedures on) and maybe have an automatic takeoff option (just like we have an autostart) instead.

**Comment by wlbragg** [comment]
>I can't understand why just clicking "Autostart" is too difficult?

😄

**Comment by tonghuix** [comment]
@onox I agree you, but if 100 % realistic FDM not mouse-friendly, a runner up solution also acceptable.

**Comment by onox** [comment]
I use the mouse (without auto coordination)

**Comment by edward-dauvergne** [comment]
> If you can fully automate this, then I've no problems with it.

Should this be a separate issue?  This needs some structural organisation on this side.  But it will benefit all parties.

> Just FYI I think SF is a product from the 90s that does/did shady things. So I'm not gonna/can't commit to FGDATA.

Just FYI, I have (or had) accounts with Gna!, Savannah, Google Code, Gitorious, Bitbucket, GitHub, GitLab and SourceForge!  I may have forgotten some earlier ones that no longer exist.  I find such discussions so 1990s ;)  They all have their benefits and disadvantages - but I know that none are perfect.

**Comment by legoboyvdlp** [comment]
FYI, I also use the mouse. Apart from needing a fair bit of rudder input which is hard to control at the same time as the aileron, I have been able to fly this aircraft for the first time - before it was unflyable as the pitch was far too sensitive! So I think it has improved already...

**Comment by onox** [comment]
> Should this be a separate issue? This needs some structural organisation on this side. But it will benefit all parties.

Yes, please open a new issue. What kind of organization are you talking about?

**Comment by dany93** [comment]
@wlbragg wrote

> @dany93 is it possible that the omission of
> 
>     I forgot to restore the aileron trim after restoring the previous value for aileron efficiency.
>     is affecting to the spiraling prop wash effect? Or does the spiraling prop wash effect feel as you intended?

My error on the aileron trim had an obvious effect in flight, rather at moderate or high speed. A rightwards rolling effect. I think that the yawing effect was negligible.
The yawing effect when rolling before takeoff (when the spiral propwash effect is important) is very secondary, IMO negligible. I think, impossible to sense compared with the spiral propwash.

**Comment by dany93** [comment]
For everyone who wish to test, even in its current version (before @wlbragg's options), you can tune the spiraling propwash effect in sim by setting a value in the Internal properties
`fdm/jsbsim/aero/coefficient/spiral-propwash-coeff`
at 1 per default. Every value (0.8 = 80%) is possible.
If you find a value which better fits your preference.... Please, pilots!

**Comment by edward-dauvergne** [comment]
I was just wondering - could it be that the propwash is correct but the P-factor has been previously set too high as a way to compensate for the lack of propwash in older versions of the aircraft?

**Comment by algefaen** [comment]
> If you find a value which better fits your preference.... Please, pilots!

Hi @dany93, I'm planning to take our club's 172N+O-360 for a trip this weekend. I suggested I could record the movement of the ball while climbing / descending at various airspeeds, without touching the rudder. Do you have any other suggestions for useful data I could try to gather? (Note that weather is always unpredictable and might make the data less useful or force me to cancel the flight, I can't promise anything.)

(I'm not going to try to take off without using rudder or other obviously dangerous ideas.:)

**Comment by dany93** [comment]
Hi @algefaen that's very kind of you!

First of all, do not take any risk, do not let any thinking make yourself distracted from the piloting.

If you can, observe the full amplitude of the rudder when still on the ground, in order to memorize it and to try having an idea of your relative action later.

Please pilot as you think you should do.  When everything goes well, if you can briefly observe your rudder movements and/or action amplitude, during going along on the runway before takeoff and during climbing, that will be a contribution.
The aim is to observe the needed action on the rudder for keeping in line on the runway and keep the ball centered at climbing at a given (not too fast but secure) velocity. In accordance with the safe manner. Even if these data are not measurable, are subjective, later based on memory. We are not machines, nor computers.

To give some indications, with the current c172 in github (`spiral-propwash-coeff` = 0.5): at rolling, I have to push the rudder pedals about 20% (briefly), then 15%...10% to keep in line on the runway. In FG, it is easy to measure in the properties.

If you can memorize and bring some data, great. If you cannot for any reason (e.g. if you have been too concentrated on other things) no worry.
In any case, thank you very much for your intention.

After that, even by mere intuition / feeling, maybe you can have a better idea for the value to set in our aircraft for the `spiral-propwash-coeff` (0.5? 0.25?). At least, il will help to find a compromise concerning this issue.

If you find something else, thanks too. Although, as simulation is not perfect, adjusting a feature is sometimes difficult without degrading another one.

I wish you a nice flight.

Dany

**Comment by algefaen** [comment]
Hi, when climing at 60kts it feels like I have to use a lot more right rudder in the real plane than in flightgear... Even with the propwash coefficient set to 1.0. Maybe the propwash is ok but that we ought to have more p-factor. 

During the take-off-roll, the initial need for a large input (as you noted) doesn't seem to be that necessary, maybe there's more inertia and friction in the real world than in the sim? ... Then during rotation, it feels like I need to add more right rudder in real life, which is why I'm thinking p-factor maybe should be a bigger contributor.

I still want to gather some more evidence though, I think I should be able to film the copilot's rudder pedals for a flight and use that to find out how much movement you need. (It might _feel_ like more than it moves, since the rudder might be pushing back, whereas my sim pedals don't.)

In general the sim feels a bit more squirrely than the real deal, with small movements having a large initial effect, while it feels like in real life there's a bit more inertia.

**Comment by callahanp** [comment]
Could the aircraft be responding to a change in position of the control
surfaces as if they had zero mass, or as if the forces on them did not
exist?  I'm not saying that the flight model is wrong in any given control
position, just that the response of the aircraft to a change in a control
input is never instantaneous.  You have to actually move the surfaces from
one position to another and that takes accelleration and decelleration on
at the beginning and end of the move.  Also, our cheap simulation yokes
work with springs and no force feedback.  When moving a spring based Yoke
forward and back for example, the force needed to move the yoke changes
noticeably as a function of the distance from the center.  I'm pretty sure
that's not realistic.  Also with the springs in play it's easy to miss the
mark and overshoot.

I wonder if it's possible to change the position of control surfaces faster
with simulation equipment than you can or would in a real plane.  Could
that affect the difference between the sim and a real plane?

I'm also wondering about the effect of play in the controls themselves.
Play would come into consideration if you were holding the position of a
control in one direction with little or no force and then moved the same
control in the opposite direction.  Except for modeled forces on the
control, would there be little no effect from the movement in the opposite
direction until the play was overcome?

Is anyone doing force feedback with Flightgear?

Without violating FAA rules, how could you instrument a real plane to
measure control surface position, control holding pressure or torque,
aircraft airspeed,  attitude and engine RPMs.  What would a pilot notice
about flying the aircraft with instrumentation in place?

-Pat

**Comment by wlbragg** [comment]
@algefaen thank you so much for this detail, we really want to make this simulation match the real thing as closely as possible. You input here is of immense value to us, thanks again!

**Comment by dany93** [comment]
Thank you @algefaen for this report.

The P-factor is an issue, more than I thought. After my tests, it seems  that it has almost no effect in FG, whatever the set value in the propeller file. I'm waiting for confirmation (or not) of this by other users. Hopefully... Unfortunately, it seems that this feature is almost ignored by the aircraft developers.
I'm in contact with the JSBSim-Team.
https://github.com/JSBSim-Team/jsbsim/issues/64#issue-332380822

Thank you too for your idea of filming the rudder movements, but I'm afraid that the accuracy of the data that you will gather cannot be exploited at their value for at least two main reasons:
- The FDM has its limits, and it's sometimes difficult to fulfill several features without degrading others (all the more for numerous, very accurate features),
- @callahanp considerations: Due to the differences between joysticks (force from a spring, proportional to the deflection) and reality (force feedback from the control surfaces + mechanical friction) the feelings are irremediably different. Which (fortunately) still enables having good sensations in simulation, with realistic responses to actions. But we have to accept this.
So, taking into account the difficulty of what you wish to do, please do as you wish but have this in mind.

But your feedback and feeling (even if subjective) will always be precious. This elastic link between reality and simulation obviously brings improvements.

**Comment by callahanp** [comment]
I know someone who built his own:

https://www.mycessnasim.info/
https://www.facebook.com/mycessnasim/videos/699826833510517/

**Comment by callahanp** [comment]
Can you tell me more about inertia and damping in the real aircraft?
How about inertia and damping in our simulated C172P?
What are the considerations?

**Comment by dany93** [comment]
P-factor issue solved with the JSBSim-team.
Implementation in https://github.com/c172p-team/c172p/issues/1130

**Comment by dany93** [comment]
@callahanp wrote:

> Can you tell me more about inertia and damping in the real aircraft?

No.
I've made a few attempts to calculate the moments of inertia for some aircraft. Once, I found almost exactly the values set in FG, I think those from aeromatic. Probably by a good contribution of luck. But in general my opinion was that the calculated values were very approximative because of the evaluations on weight distributions.
Those evaluations make the calculations quite difficult.
Fortunately, in the past (don't remember when...), I've done a few tests by changing the moments of inertia, and found that the effect was weak. A high accuracy does not not seem mandatory.

> How about inertia and damping in our simulated C172P?

- Inertia:
The initial values were those probably given by Megginson. They were in accordance with those in his [linear aerodynamic coefficients table](http://www.megginson.com/Aviation/roskam-coefficients.html#sec.data) for the Cessna 182.
They have been very slightly changed after @okroth [advices](https://github.com/c172p-team/c172p/issues/1047#issuecomment-381964443), to take into account weight differences and distribution.


- Damping:
The coefficients are in accordance with Megginson's [linear aerodynamic coefficients table](http://www.megginson.com/Aviation/roskam-coefficients.html#sec.data) for the Cessna 182. I don't remember having changed them.
